lib/ecto/changeset.ex

```diff
@@ -1902,6 +1888,7 @@ defmodule Ecto.Changeset do
       value when is_binary(value) and trim -> String.trim_leading(value) == ""
       value when is_binary(value) -> value == ""
       nil -> true
+      [] -> true
```

Comment:
I am worried about the implication of this change for the general usage of `{:array, any_type}`. 🤔

Reply:
Agreed, if I am reading it correctly, there is a difference in functionality between "required: true" on `cast_embed` (which fails on `[]`) and `validate_required` on an `{:array, any_type}`, which does not fail if it is a `[]`. If I remove the `[] -> true` clause, [this line fails](https://github.com/elixir-ecto/ecto/blob/master/test/ecto/changeset/embedded_test.exs#L467)